src/SAWScript/Options.hs

```diff
@@ -74,66 +83,77 @@ printOutWith setting level msg
 printOutLn :: Options -> Verbosity -> String -> IO ()
 printOutLn o v s = printOutFn o v (s ++ "\n")
 
-options :: [OptDescr (Options -> Options)]
+options :: [OptDescr (Options -> IO Options)] -- added IO to do validation here instead of later
```

Comment:
At some point we might want to switch to a "real" option parsing library so that we can handle things like defaults, option groups, better validation, and useful error messages in a more principled way.

This seems fine to me for now, though.